sys/callout.c

```diff
@@ -90,5 +91,12 @@ void callout_process(realtime_t time) {
 
       elem = next;
     }
+    if (current_bucket == last_bucket)
+      break;
+
+    current_bucket = (current_bucket + 1) % CALLOUT_BUCKETS;
   }
+
+  /* Update ci.last. */
```

Comment:
Yeah, what I actually wanted to express was something along the lines of "`ci.last` was not updated above".